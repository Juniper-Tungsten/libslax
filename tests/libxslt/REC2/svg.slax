/* Machine Crafted with Care (tm) by slaxWriter */
version 1.0;

ns "http://www.w3.org/Graphics/SVG/SVG-19990812.dtd";

output-method xml {
    indent "yes";
    media-type "image/svg";
}

match / {
    <svg width="3in" height="3in"> {
        <g style="stroke: #000000"> {
            /* draw the axes */
            <line x1="0" x2="150" y1="150" y2="150">;
            <line x1="0" x2="0" y1="0" y2="150">;
            <text x="0" y="10"> "Revenue";
            <text x="150" y="165"> "Division";
            
            for-each (sales/division) {
                /* define some useful variables */
                /* the bar's x position */
                var $pos = (position() * 40) - 30;
                /* the bar's height */
                var $height = revenue * 10;
                /* the rectangle */
                <rect x=$pos y=150 - $height width="20" height=$height>;
                /* the text label */
                <text x=$pos y="165"> @id;
                /* the bar value */
                <text x=$pos y=145 - $height> revenue;
            }
        }
    }
}
