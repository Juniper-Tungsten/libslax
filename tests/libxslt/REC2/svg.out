<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/Graphics/SVG/SVG-19990812.dtd" width="3in" height="3in">
  <g style="stroke: #000000">
    <line x1="0" x2="150" y1="150" y2="150"/>
    <line x1="0" x2="0" y1="0" y2="150"/>
    <text x="0" y="10">Revenue</text>
    <text x="150" y="165">Division</text>
  </g>
</svg>
